src/rdkafka_mock_handlers.c

```diff
@@ -872,7 +888,22 @@ rd_kafka_mock_buf_write_Metadata_Topic(rd_kafka_mock_cluster_t *mcluster,
         /* Response: Topics.ErrorCode */
         rd_kafka_buf_write_i16(resp, err);
         /* Response: Topics.Name */
-        rd_kafka_buf_write_str(resp, topic, -1);
+        if (ApiVersion >= 12 && err == RD_KAFKA_RESP_ERR_UNKNOWN_TOPIC_ID) {
+                rd_kafka_buf_write_str(resp, NULL, -1);
+        } else {
+                rd_kafka_buf_write_str(resp, topic, -1);
+        }
+
+        if (ApiVersion >= 10) {
+                if (mtopic) {
+                        rd_kafka_Uuid_t topic_id = mtopic->id;
+                        /* Response: Topics.TopicId */
+                        rd_kafka_buf_write_uuid(resp, &topic_id);
+                } else {
+                        rd_kafka_buf_write_uuid(resp, &RD_KAFKA_UUID_ZERO);
+                }
+        }
```

Comment:
When requesting by topic id, we should always return requested topic id in the response instead of ZERO_UUID.